### ISSUE TITLE
Azure Provisioner - Authorize based on Subscription ID

### ISSUE THREAD

**Issue by rvichery** [smallstep/certificates#735]
### What would you like to be added
Today, Azure provisioner requests are authorized based on Tenant ID and optionally Resource Groups. I think it would be good to also support filtering/authorizing base on Subscription ID. The Subscription ID is available in the Azure Token under `XMSMirID` path.

### Why this is needed
In my company's environment, Resource Groups are dynamically created/deleted (multiple times a day) so configuring Resource Group filtering at the provisioner level is not an option. Our Tenant ID contains many subscriptions (dev, staging, pre-prod, prod, ...), which will let unwanted Virtual Machines issue certificate against a given intermediate CA (dev VM to Prod Intermediate CA for example).

I would be happy to contribute this feature/enhancement.

**Comment by maraino** [comment]
Good point, as you say we currently only support filtering by resource groups, but not subscription ids, but the logic should be pretty similar. We'll bring this to our next triage meeting for prioritization, if I don't fix it before that :)

**Comment by maraino** [comment]
Hi @rvichery, I'm seeing now that you're willing to contribute this feature, if you do, make sure to add the functionality in all places where ResourceGroup appears:
* authority/provisioner/azure.go
* authority/provisioner/azure_test.go
* authority/provisioners.go

You will also need to change the [linkedca repo](https://github.com/smallstep/linkedca/blob/708f4c5b5a8917a2d31991ea541c19805e26a9fd/provisioners.proto#L113-L119), and we would like to add that functionality to the `step` cli too.

**Comment by rvichery** [comment]
@maraino I can probably get a PR going over the weekend for this. Thank you for pointing me to the https://github.com/smallstep/linkedca/blob/708f4c5b5a8917a2d31991ea541c19805e26a9fd/provisioners.proto#L113-L119. I will also make the changes in `step` cli.

**Comment by rvichery** [comment]
Sorry for the delay on this one, I have been quite busy with other projects. I looked at how to implement this and was faced with a fundamental question on how to handle cases where a list of subscription IDs and a list of resource groups are configured: Should we consider a logical disjunction between the two lists where allowing the token? 
Another option could be to enhance the way resource groups are configured today and have the following format options:
- `subscriptionA/*` to allow all resource group in the subscription A
- `subscriptionA/resourceGroupA` to only allow resource group A in subscription A
- `resourceGroupA` to only allow resource group A from all subscriptions (backward compatible with today's behavior)

Please tell me what you think and I can implement accordingly.
